test/ideal/freealg-test.jl

```diff
@@ -0,0 +1,25 @@
+@testset "freealg" begin
+  R, (u11,u12,u13,u14,
+  u21,u22,u23,u24,
+  u31,u32,u33,u34,
+  u41,u42,u43,u44) = FreeAlgebra(QQ, ["u11", "u12", "u13", "u14",
+  "u21", "u22", "u23", "u24",
+  "u31", "u32", "u33", "u34",
+  "u41", "u42", "u43", "u44"], 7)
+  #rs1 = u11 + u12 + u13 + u14 - 1
+  rs2 = u21 + u22 + u23 + u24 - 1
+  rs3 = u31 + u32 + u33 + u34 - 1
+  rs4 = u41 + u42 + u43 + u44 - 1
+  cs1 = u11 + u21 + u31 + u41 - 1
+  cs2 = u12 + u22 + u32 + u42 - 1
+  cs3 = u13 + u23 + u33 + u43 - 1
+  cs4 = u14 + u24 + u34 + u44 - 1
+  
+  Singular.libSingular.set_option("OPT_REDTAIL",false,R.ptr)
```

Comment:
It is good that these specific user cases are adjusted.
You adjust the code in `ideal.jl` and `module.jl` to use the new `set_option` variant, and also use that in this test.

But what about `set_option` calls in the `with_redTail` helper and other similar helpers? Those are not adjusted, are they?

In particular, I don't see how this PR fixes issue #808 -- it provides a way to avoid the issue, which of course is good. But I don't see how it resolve the issues listed there specifically?

Reply:
One approach to actually resolve the issue would be to change `with_redTail` (and its siblings) to require two argument: first the ring, then the option value. Of course that would be a breaking change, but one that it is hopefully fairly easy to adapt to.

Reply:
This test should be adapted to use the new `with_redTail` helper